C#/C#.sublime-syntax

```diff
@@ -1,278 +1,1634 @@
-            - match: "(?=})"
+            1: entity.name.section.cs
+        - include: option_done
+    - match: '(endregion)\s*'
+      scope: storage.type.section.cs
+      push:
+        - match: ([^/\s][^/]*?)\s*(?=//|/\*|$)
+          captures:
+            1: variable.other.section.cs
+        - include: option_done
+
+    - match: '(line)\s+(default|hidden)'
+      captures:
+        1: keyword.other.preprocessor.cs
+        2: keyword.other.preprocessor.cs
+      push: option_done
+    - match: '(line)\s*([0-9]*)\s*(".*")?'
+      captures:
+        1: keyword.other.preprocessor.cs
+        2: constant.numeric.cs
+        3: string.quoted.double.cs
+      push: option_done
+    - match: '(pragma\s+checksum)\s+(".*")\s+("\{)(.*)(\}")\s+("\{)(.*)(\}")'
+      captures:
+        1: keyword.other.preprocessor.cs
+        2: string.quoted.double.cs
+        3: punctuation.definition.hash.begin.cs
+        4: constant.numeric.cs
+        5: punctuation.definition.hash.begin.cs
+        6: punctuation.definition.hash.begin.cs
+        7: constant.numeric.cs
+        8: punctuation.definition.hash.begin.cs
+      push: option_done
+
+    - match: '(pragma)'
+      scope: keyword.other.preprocessor.cs
+      push: option_done
+
+    - match: '(.*)'
+      scope: invalid.illegal.cs
+    - match: $
+      pop: true
+
+  option_message:
+    - meta_content_scope: string.unquoted.cs
+    - include: comments
+    - match: $
+      pop: true
+
+  # option_done pops out at the end of the line and handles comments
+  option_done:
+    - include: comments
+    - match: $
+      pop: true
+    - match: \S
+      scope: invalid.illegal.cs
+
+  main:
+    - match: '\}'
+      scope: invalid.illegal.unexpected.closing-bracket
+
+    - match: '\{'
+      scope: punctuation.section.block.begin.cs
+      push:
+        - match: '\}'
+          scope: punctuation.section.block.end.cs
+          pop: true
+        - include: main
+
+    # skips opening
+    - match: ^\s*
+
+    - include: tag
+    - include: using
+    - include: class_declaration
+    - include: interface_declaration
+    - include: namespace_declaration
+    - include: delegate_declaration
+    # - include: method_declaration
+
+    # allows coloration of code outside a class
+    - include: code_block_in
+
+  using:
+    - match: '\b(using)\s+{{name}}\s*(=)\s*'
+      captures:
+        1: keyword.control.import.cs
+        2: entity.name.type.cs
+        3: storage.modifier.namespace.as.cs
+      push: using_namespace
+    - match: '\b(using)\s+(static\b)?\s*{{name}}'
+      captures:
+        1: keyword.control.import.cs
+        2: storage.modifier.namespace.cs
+        3: variable.namespace.cs
+      push: using_namespace
+
+  using_namespace:
+    - match: '{{name}}'
+      scope: variable.namespace.cs
+    - match: \.
+      scope: punctuation.separator.namespace.cs
+    - match: '<'
+      scope: punctuation.definition.arguments.type.begin.cs
```

Comment:
Is this a generic? If so, consider using `punctuation.definition.generic`. If not, `punctuation.section` probably makes more sense.